sdk/anomalydetector/azure-ai-anomalydetector/src/samples/java/com/azure/ai/anomalydetector/MultivariateSample.java

```diff
@@ -85,105 +91,103 @@ private static AnomalyDetectorClient getClient(String endpoint, String key) {
-        RequestOptions requestOptions = new RequestOptions();
-        Response<BinaryData> response = client.detectMultivariateLastAnomalyWithResponse(modelId.toString(), body, requestOptions);
-
-        return response;
+    private static LastDetectionResult getLastDetectResult(AnomalyDetectorClient client, LastDetectionRequest body, UUID modelId) {
+        LastDetectionResult res = client.detectMultivariateLastAnomaly(modelId.toString(), body);
+        return res;
     }
 
-    public static void run(BinaryData trainBody, BinaryData beginInferBody) throws Exception {
+    public static void run(String datasource, DataSchema dataSchema) throws Exception {
         String endpoint = "<anomaly-detector-resource-endpoint>";
         String key = "<anomaly-detector-resource-key>";
 
         // Get multivariate client
         AnomalyDetectorClient client = getClient(endpoint, key);
 
+        // set training request
+        OffsetDateTime startTime = OffsetDateTime.of(2021, 1, 2, 0, 0, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime endTime = OffsetDateTime.of(2021, 1, 2, 5, 0, 0, 0, ZoneOffset.UTC);
+        ModelInfo trainRequest = new ModelInfo(datasource, startTime, endTime);
+        trainRequest.setSlidingWindow(200);
+        AlignPolicy alignPolicy = new AlignPolicy();
+        alignPolicy.setAlignMode(AlignMode.OUTER);
+        alignPolicy.setFillNAMethod(FillNAMethod.LINEAR);
+        alignPolicy.setPaddingValue(0.0);
+        trainRequest.setAlignPolicy(alignPolicy);
+        trainRequest.setDataSchema(dataSchema);
+        trainRequest.setDisplayName("SampleRequest");
+
         // Start training and get Model ID
-        UUID modelId = createModel(client, trainBody);
-        System.out.println(modelId);
+        UUID modelId = createModel(client, trainRequest);
+        System.out.println("modelId: " + modelId);
         // Check model status util the model get ready
         while (true) {
-            JsonObject modelInfo = getModelInfo(client, modelId);
-            ModelStatus modelStatus = ModelStatus.valueOf(modelInfo.getString("status"));
+            ModelInfo modelInfo = getModelInfo(client, modelId);
+            ModelStatus modelStatus = ModelStatus.valueOf(modelInfo.getStatus().toString());
             if (modelStatus == ModelStatus.READY) {
                 System.out.println("READY");
                 break;
             } else if (modelStatus == ModelStatus.FAILED) {
                 System.out.println("FAILED");
-                throw new Exception(modelInfo.getJsonArray("errors").getString(0));
+                String errorStr = "";
+                for (ErrorResponse errorResponse : modelInfo.getErrors()) {
+                    System.out.println(errorResponse.getCode() + errorResponse.getMessage());
+                    errorStr += ";" + errorResponse.getCode() + errorResponse.getMessage();
+                }
+                throw new Exception(errorStr);
```

Comment:
Should not throw bare `Exception` in Java.

At least use a `RuntimeException`, or proper subclass of it.

Reply:
fixed.